addons/sale_stock/models/sale_order.py

```diff
@@ -42,6 +47,28 @@ def _default_warehouse_id(self):
     json_popover = fields.Char('JSON data for the popover widget', compute='_compute_json_popover')
     show_json_popover = fields.Boolean('Has late picking', compute='_compute_json_popover')
 
+    def _init_column(self, column_name):
+        """ Ensure the default warehouse_id is computed without considering the company_dependent field on res_users
+
+        At column initialization, the ir.model.fields for res.users.property_warehouse_id isn't created,
+        which means trying to read the field to get the default value will crash.
+        We therefore enforce the default here, without going through
+        the default function on the warehouse_id field.
+        """
+        if column_name != "warehouse_id":
+            super(SaleOrder, self)._init_column(column_name)
+        else:
+            field = self._fields[column_name]
+            default = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
```

Comment:
`_init_column` is invoked with the superuser (uid=1) and without any company in context.
Therefore, `self.env.company` is the superuser's company.
Are you sure that it is what you want?

Reply:
It is already done like this, defaults for new columns on existing tables are wrong in multi-company...